homeassistant/components/evohome/__init__.py

```diff
@@ -47,8 +48,13 @@
 
 
 def _local_dt_to_utc(dt_naive: datetime) -> datetime:
```

Comment:
Now that we have found a solution for conversion, we should be able to ditch these functions now too?

Reply:
Actually, I was going to keep them (they're used elsewhere), albeit using the 'solution':
```python
def _local_dt_to_utc(dt_naive: datetime) -> datetime:
    dt_aware = utcnow() + (dt_naive - datetime.now())
    return dt_aware.replace(microsecond=0, tzinfo=None)


def _utc_to_local_dt(dt_naive: datetime) -> datetime:
    dt_aware = as_local(dt_naive)
    return dt_aware.replace(microsecond=0, tzinfo=None)
```
... but I can't get it to work!

Reply:
Why are local datetimes even needed? We should never do anything with local datetimes.

Reply:
In the first example, you're returning a UTC datetime but remove tz, which you probably don't want. 

Second one, should be fine ?

Reply:
> Why are local datetimes even needed?

The local datetime is used by the client (when I substantiate it) to validate the tokens retrieved from storage.

The two converts are correct - I found the problem, I needed to add `.replace(tzinfo=UTC)`:
```python
        access_token_expires_utc = _local_dt_to_utc(
            self.client.access_token_expires)
...
        self._app_storage[CONF_ACCESS_TOKEN_EXPIRES] = \
            access_token_expires_utc.isoformat()
...
        async_track_point_in_utc_time(
            self.hass,
            self._save_auth_tokens,
            access_token_expires_utc.replace(tzinfo=UTC)
        )
```

Reply:
Must I store datetime strings with offsets?

Reply:
We should only store UTC in the state attributes, or even better, use device class datetime. That way users can render it as they want (countdown, etc)

Reply:
OK, stored as UTC now.